ui/app/src/model/dashboard-client.ts

```diff
@@ -0,0 +1,25 @@
+// Copyright 2022 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { useQuery } from '@tanstack/react-query';
+import { DashboardResource, fetchJson } from '@perses-dev/core';
```

Comment:
isn't that weird `DashboardResource` is part of `@perses-dev/core` and not part of `@perses-dev/dashboard` ?

Reply:
hmm it's been awhile but I think we did that to get around a circular dependency at one point

Reply:
I've basically been putting all the "core models" for all resources that Perses framework knows about into `core`. I think the main reason was because the `plugin-system` also needs to know about some of those types.